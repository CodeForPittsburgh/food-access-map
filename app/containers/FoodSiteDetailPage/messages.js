/*
 * FoodSiteDetailPage Messages
 *
 * This contains all the text for the FoodSiteDetail container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.FoodSiteDetailPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Detail View',
  },
});
