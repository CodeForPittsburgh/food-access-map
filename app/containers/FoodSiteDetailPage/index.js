/*
 * FoodSiteDetailPage
 *
 * This is a details page for food sites
 *
 */

import React from 'react';
import Grid from '@material-ui/core/Grid';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import messages from './messages';

export default function FoodSiteDetailPage(props) {
  return (
    <Grid
      container
      direction="column"
      justify="space-around"
      alignItems="center"
    >
      <Link to="/">
        I go back home! Oh god why have you not made me a dedicated nav
        component yet?!
      </Link>
      <h1 className="headline">
        <FormattedMessage {...messages.header} />
      </h1>
      <p>
        oh hey I am the detail page for now. Fill me with styled display
        components
      </p>
      <p>This is my uniq id! behold its glory!!! {props.match.params.uniqId}</p>
    </Grid>
  );
}

FoodSiteDetailPage.propTypes = {
  match: PropTypes.object.isRequired,
};
