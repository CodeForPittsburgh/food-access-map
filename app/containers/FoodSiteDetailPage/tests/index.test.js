import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import FoodSiteDetailPage from '../index';

const renderer = new ShallowRenderer();

const fakeId = {
  params: {
    uniqId: 123,
  },
};

describe('<FoodSiteDetailPage />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<FoodSiteDetailPage match={fakeId} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
