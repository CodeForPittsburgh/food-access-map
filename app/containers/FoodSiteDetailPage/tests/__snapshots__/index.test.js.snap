// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodSiteDetailPage /> should render and match the snapshot 1`] = `
<ForwardRef(WithStyles)
  alignItems="center"
  container={true}
  direction="column"
  justify="space-around"
>
  <Link
    to="/"
  >
    I go back home! Oh god why have you not made me a dedicated nav component yet?!
  </Link>
  <h1
    className="headline"
  >
    <FormattedMessage
      defaultMessage="Details Page (better title . . .)"
      id="app.containers.FoodSiteDetailPage.header"
      values={Object {}}
    />
  </h1>
  <p>
    oh hey I am the detail page for now. Fill me with styled display components
  </p>
  <p>
    This is my uniq id! behold its glory!!! 
    123
  </p>
</ForwardRef(WithStyles)>
`;
